omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java

```diff
@@ -0,0 +1,148 @@
+        return new NeedsPaging<LogEntry>(
+                Context.getService(IdentifierSourceService.class).getLogEntries(null, null, null, null, null, null),
+                context);
+    }
+
+    @Override
+    protected PageableResult doSearch(RequestContext context) {
+        IdentifierSourceService identifierSourceService = Context.getService(IdentifierSourceService.class);
+        UserService service = Context.getUserService();
+
+        String source = context.getRequest().getParameter("source");
+        String fromDate = context.getRequest().getParameter("fromDate");
+        String toDate = context.getRequest().getParameter("toDate");
+        String identifier = context.getRequest().getParameter("identifier");
+        String comment = context.getRequest().getParameter("comment");
+        String generatedBy = context.getRequest().getParameter("generatedBy");
+
+        List<LogEntry> logEntries = new ArrayList<LogEntry>();
+
+        IdentifierSource logSource = source != null ? identifierSourceService.getIdentifierSourceByUuid(source) : null;
+        Date dateFrom = fromDate != null ? (Date) ConversionUtil.convert(fromDate, Date.class) : null;
+        Date dateTo = toDate != null ? (Date) ConversionUtil.convert(toDate, Date.class) : null;
+        User user = generatedBy != null ? service.getUserByUuid(generatedBy) : null;
+        if (logSource == null && identifier == null && comment == null && dateFrom == null && dateTo == null
+                && user == null) {
+            return new EmptySearchResult();
+        }
+        logEntries = identifierSourceService.getLogEntries(logSource, dateFrom, dateTo, identifier, user, comment);
+
+        return new NeedsPaging<LogEntry>(logEntries, context);
+    }
+
+    @Override
+    public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+        DelegatingResourceDescription description = null;
+        if (rep instanceof RefRepresentation) {
+            description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("display");
+            description.addProperty("identifier");
+            description.addSelfLink();
+        } else if (rep instanceof DefaultRepresentation || rep instanceof FullRepresentation) {
+            description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("source");
+            description.addProperty("identifier");
+            description.addProperty("comment");
+            description.addProperty("generatedBy");
+            description.addProperty("dateGenerated");
+            description.addSelfLink();
+            if (rep instanceof DefaultRepresentation) {
+                description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+            }
+        }
+        return description;
+    }
+
+    @PropertyGetter("uuid")
```

Comment:
I'm curious why you have to use methods with @PropertyGetter for all these  properties? Don't they have actual getters on the LogEntry class?

Reply:
Ive removed the @PropertyGetter for the uuid. The _source_ and _generatedBy_ getter methods the return objects which were creating errors so I am using the @PropertyGetter to get the uuids. Also there is no getter for the display